src/libraries/Pool.sol

```diff
@@ -237,18 +237,12 @@ library Pool {
         }
     }
 
-    // the top level state of the swap, the results of which are recorded in storage at the end
-    struct SwapState {
-        // the amount remaining to be swapped in/out of the input/output asset
-        int256 amountSpecifiedRemaining;
-        // the amount already swapped out/in of the output/input asset
-        int256 amountCalculated;
-        // current sqrt(price)
+    // the results after a swap, written to storage and returned
```

Comment:
i found this misleading. It doesnt just hold the results after a swap, but during the entire swap too.
I'd rather it just said like
```
Tracks the state of a pool throughout a swap, and returns the resultant values at the end of the swap
```
or something?
I do think `SwapState` is much more descriptive of what this is. `SwapResult` implies it holds the result of a swap... which would be the unspecified token and final price i guess? Really its tracking the pool's state throughout the whole swap

Reply:
Yeah I just can't think of something better & that doesn't overload state. can update comment